psiminer-cli/src/main/kotlin/config/LabelExtractorConfigs.kt

```diff
@@ -6,12 +6,12 @@ import labelextractor.MethodNameLabelExtractor
 import labelextractor.LabelExtractor
 
 @Serializable
-abstract class LabelExtractorConfig {
-    abstract fun createProblem(): LabelExtractor
+abstract class LabelExtractorConfig<T> {
+    abstract fun createProblem(): LabelExtractor<T>
```

Comment:
Basically, the config shouldn't care about what type of data users want to extract. Its goal is to determine pipeline steps. Therefore, it's quite strange to have type parametrization here. All information that should be kept here is that we can create an extractor to extract something. Best way to do it is to use star-projection:
```kotlin
@Serializable
abstract class LabelExtractorConfig {
    abstract fun createProblem(): LabelExtractor<*>
}
```